application/models/QuestionTheme.php

```diff
@@ -1087,4 +1093,77 @@ public static function getDummyInstance($questionType)
         $questionTheme->settings = $settings;
         return $questionTheme;
     }
+
+    /**
+     * Returns the type of question theme (coreQuestion, customCoreTheme, customUserTheme)
+     *
+     * coreQuestion = Themes shipped with Limesurvey that don't extend other theme
+     * customCoreTheme = Themes shipped with Limesurvey that extend other theme
+     * customUserTheme = User provided question themes
+     *
+     * @return string
+     */
+    public function getThemeType()
+    {
+        if ($this->core_theme) {
+            if (empty($this->extends)) {
+                return self::THEME_TYPE_CORE;
+            } else {
+                return self::THEME_TYPE_CUSTOM;
+            }
+        } else {
+            return self::THEME_TYPE_USER;
+        }
+    }
+
+    /**
+     * Returns the XML path relative to the path configured for the question theme type
+     * @return string
+     */
+    public function getRelativeXmlPath()
+    {
+        $type = $this->getThemeType();
+        $typeDirectory = self::getQuestionThemeDirectoryForType($type);
+
+        // xml_path is supposed to contain the type directory, so we extract the rest of the path
+        $relativePath = substr($this->xml_path, strpos($this->xml_path, $typeDirectory) + strlen($typeDirectory));
+        $relativePath = ltrim($relativePath, "\\/");
+        return $relativePath;
+    }
+
+    /**
+     * Returns the XML path
+     * It may be absolute or relative to the Limesurvey root
```

Comment:
Hm, it should always be relative, but question theme is relative config userquestionthemerootdir, not the LS root. Right?

Reply:
https://github.com/LimeSurvey/LimeSurvey/blob/26349710d4885c58ef542e80b8dae93d7afb9514/application/config/config-defaults.php#L737